dask/array/random.py

```diff
@@ -1,56 +1,284 @@
 from __future__ import absolute_import, division, print_function
 
 import numpy as np
-from .wrap import wrap, wrap_func_size_as_kwarg
+from itertools import product
+from toolz import curry
+from .core import normalize_chunks, Array, names
 
-"""
-Univariate distributions
-"""
+def doc_wraps(func):
+    """ Copy docstring from one function to another """
+    def _(func2):
+        func2.__doc__ = func.__doc__.replace('>>>', '>>').replace('...', '..')
+        return func2
+    return _
 
-wrap = wrap(wrap_func_size_as_kwarg)
-
-random = wrap(np.random.random)
-beta = wrap(np.random.beta)
-binomial = wrap(np.random.binomial)
-chisquare = wrap(np.random.chisquare)
-exponential = wrap(np.random.exponential)
-f = wrap(np.random.f)
-gamma = wrap(np.random.gamma)
-geometric = wrap(np.random.geometric)
-gumbel = wrap(np.random.gumbel)
-hypergeometric = wrap(np.random.hypergeometric)
-laplace = wrap(np.random.laplace)
-logistic = wrap(np.random.logistic)
-lognormal = wrap(np.random.lognormal)
-logseries = wrap(np.random.logseries)
-negative_binomial = wrap(np.random.negative_binomial)
-noncentral_chisquare = wrap(np.random.noncentral_chisquare)
-noncentral_f = wrap(np.random.noncentral_f)
-normal = wrap(np.random.normal)
-pareto = wrap(np.random.pareto)
-poisson = wrap(np.random.poisson)
-power = wrap(np.random.power)
-rayleigh = wrap(np.random.rayleigh)
-triangular = wrap(np.random.triangular)
-uniform = wrap(np.random.uniform)
-vonmises = wrap(np.random.vonmises)
-wald = wrap(np.random.wald)
-weibull = wrap(np.random.weibull)
-zipf = wrap(np.random.zipf)
 
-"""
-Standard distributions
-"""
+class RandomState(object):
+    def __init__(self, seed=None):
+        self.state = np.random.RandomState(seed)
 
-standard_cauchy = wrap(np.random.standard_cauchy)
-standard_exponential = wrap(np.random.standard_exponential)
-standard_gamma = wrap(np.random.standard_gamma)
-standard_normal = wrap(np.random.standard_normal)
-standard_t = wrap(np.random.standard_t)
+    def wrap(self, func, *args, **kwargs):
+        size = kwargs.pop('size')
+        chunks = kwargs.pop('chunks')
 
-"""
-TODO: Multivariate distributions
+        if not isinstance(size, (tuple, list)):
+            size = (size,)
+
+        chunks = normalize_chunks(chunks, size)
+        name = next(names)
+
+        # Get dtype
+        kw = kwargs.copy()
+        kw['size'] = (0,)
+        dtype = func(np.random.RandomState(), *args, **kw).dtype
+
+        # Build graph
+        keys = product([name], *[range(len(bd)) for bd in chunks])
+        sizes = product(*chunks)
+        vals = ((apply_random, func, self.state.randint(2**31),
```

Comment:
I think you can use `2 ** 32 - 1` or even better: `np.iinfo(np.uint32).max` here.